drag_and_drop_v2/public/js/drag_and_drop.js

```diff
@@ -1917,16 +1917,16 @@ function DragAndDropBlock(runtime, element, configuration) {
     };
 
     var isPastDue = function () {
-        return !configuration.has_deadline_passed;
+        return configuration.has_deadline_passed;
     };
 
-    var canShowAnswer = function() {
-        return configuration.mode === DragAndDropBlock.ASSESSMENT_MODE && !attemptsRemain();
-    };
 
     var attemptsRemain = function() {
         return !configuration.max_attempts || configuration.max_attempts > state.attempts;
     };
+    var canShowAnswer = function() {
+        return state.answer_available || configuration.answer_available;
```

Comment:
Nit: we should use `configuration.answer_available` only when `state.answer_available` is undefined. Otherwise, we should rely solely on `state.answer_available`.